### PR TITLE
Some navigation improvements

### LINKED ISSUE #262: Cancel navigation duplicates surfaces
### Expected behaviour

Surfaces are not duplicated if I cancelled navigation.

### Actual behaviour

Surfaces are duplicated when cancelling navigation leading to problems like duplicated id, etc.

### Steps to reproduce

1. Create an app with two pages (PageA and PageB) in a such a way that you navigate between them using senna.js
2. Being on PageA, click on link to navigate to PageB.
3. Quickly, before PageB is loaded, click again to navigate to PageA.
4. Inspect the DOM and you'll see surfaces duplicated.

### Browsers affected

It's easier to reproduce on IE11 because is slower, but you can use CPU throttling in Chrome.

### PR DESCRIPTION
**Pull request by diegonvs**

### CODE REVIEW

**Comment by diegonvs** [comment]
Hey @brunobasto and @jbalsas, could you guys review this?

File: src/app/App.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 import { addClasses, delegate, match, on, removeClasses } from 'metal-dom';
-import { array, async, isDefAndNotNull, isString } from 'metal';
+import { array, async, object, isDefAndNotNull, isString } from 'metal';
```
Comment (jbalsas) [inline src/app/App.js]: Can you move this to the end of the list? I think we do sort this in general :)

File: src/app/App.js [hunk 2]
```diff
@@ -123,6 +123,14 @@ class App extends EventEmitter {
 		 */
 		this.nativeScrollRestorationSupported = ('scrollRestoration' in globals.window.history);
 
+		/**
+		 * When set to true means that the current navigation cannot be Cancelled to start another.
+		 * @type {boolean}
+		 * @default false
+		 * @protected
+		 */
+		this.isNavigationAvoided = false;
```
Comment (jbalsas) [inline src/app/App.js]: I find this variable a bit confusing... could we try to figure out a different option?

If we stay with boolean vars, other options could be:
- `preventNavigation`
- `scheduleNavigation`
- ...

Another option, maybe more descriptive could be something like `navigationStrategy`, which could be:
- `'immediate'`: starts navigation right away (default value)
- `'schedule'`: queues all navigation events to trigger them in order after the previous one completes (we don't need to implement this option for now)
- `'scheduleLast'`: queues navigation events, keeping only the last one and triggers them when the previous one completes

Comment (brunobasto) [reply src/app/App.js]: Hey Chema, I liked the navigationStrategy idea.

File: src/app/App.js (hunk 1 above)
Comment (diegonvs) [reply src/app/App.js]: Of course! 😄

File: src/app/App.js (hunk 2 above)
Comment (diegonvs) [reply src/app/App.js]: Thats good too! I'll take a look at this!

**Comment by jbalsas** [comment]
Pull request submitted to https://github.com/liferay/senna.js/pull/265. [See changes here.](https://github.com/liferay/senna.js/compare/88a9f08~1...88a9f08) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**